planner/core/src/test/java/eu/seaclouds/platform/planner/core/DamGeneratorTest.java

```diff
@@ -124,5 +125,112 @@ public void testGroupsAsTopologyChild() throws Exception {
         assertTrue(topologyGroups.containsKey("sla_gen_info"));
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testNuroDAM() throws Exception {
+        String adp = new Scanner(new File(Resources.getResource("nuro_adp.yml").toURI())).useDelimiter("\\Z").next();
+
+        when(fakeAgreementManager.generateAgreeemntId(((Map<String, Object>) anyObject())))
+                .thenReturn(FAKE_AGREEMENT_ID);
+
+        DamGenerator damGenerator = new DamGenerator(MONITOR_URL, MONITOR_PORT, SLA_ENDPOINT, INFLUXDB_URL, INFLUXDB_PORT);
+        damGenerator.setAgreementManager(fakeAgreementManager);
+        dam = damGenerator.generateDam(adp);
+        template = (Map<String, Object>) yamlParser.load(dam);
+
+        String expectedDAMString = new Scanner(new File(Resources.getResource("nuro_dam.yml").toURI())).useDelimiter("\\Z").next();
+        Map<String, Object> expectedDAM = (Map<String, Object>) yamlParser.load(expectedDAMString);
+
+        assertNotNull(template);
+
+        Map<String, Object> generatedTopologyTemplate = (Map<String, Object>) template.get(DamGenerator.TOPOLOGY_TEMPLATE);
+        Map<String, Object> expectedTopologyTemplate = (Map<String, Object>) expectedDAM.get(DamGenerator.TOPOLOGY_TEMPLATE);
+
+        assertEquals(generatedTopologyTemplate.get("www"), expectedTopologyTemplate.get("www"));
```

Comment:
Probably it is better to compare the `template` elements with the values and to avoid using the expected DAM (`expectedTopology`). It is more clear.